OpenRA.Mods.Common/Widgets/Logic/DirectConnectLogic.cs

```diff
@@ -39,6 +39,8 @@ public DirectConnectLogic(Widget widget, Action onExit, Action openLobby, Connec
 				portField.Text = text.Substring(last + 1);
 			}
 
+			panel.Get<ButtonWidget>("JOIN_BUTTON").IsDisabled = () => string.IsNullOrEmpty(ipField.Text);
```

Comment:
It's more optimal to reuse the search

```suggestion
			var joinButton = panel.Get<ButtonWidget>("JOIN_BUTTON");
			joinButton.IsDisabled = () => string.IsNullOrEmpty(ipField.Text);
```

and reuse it on `OnClick`